protocol/test/forge/nexus/Relic.t.sol

```diff
@@ -0,0 +1,1175 @@
+        {
+            vm.startPrank(operator);
+            shard.mintBatch(bob, shardIds, amounts);
+            changePrank(bob);
+            shard.setApprovalForAll(address(relic), true);
+            relic.batchEquipShards(RELIC_1_ID, shardIds, amounts);
+
+            changePrank(executor);
+            relic.setBlacklistedShards(bob, RELIC_1_ID, shardIds, amounts);
+        }
+
+        vm.expectRevert(abi.encodeWithSelector(CommonEventsAndErrors.InvalidAddress.selector));
+        relic.unsetBlacklistedShards(address(0), RELIC_1_ID, shardIds, amounts);
+        
+        vm.expectRevert(abi.encodeWithSelector(IERC721A.URIQueryForNonexistentToken.selector));
+        relic.unsetBlacklistedShards(bob, 100, shardIds, amounts);
+
+        // invalid owner
+        vm.expectRevert(abi.encodeWithSelector(CommonEventsAndErrors.InvalidParam.selector));
+        relic.unsetBlacklistedShards(alice, RELIC_1_ID, shardIds, amounts);
+
+        amounts[0] = 3;
+        vm.expectRevert(abi.encodeWithSelector(CommonEventsAndErrors.InvalidParam.selector));
+        relic.unsetBlacklistedShards(bob, RELIC_1_ID, shardIds, amounts);
+
+        amounts[0] = 1;
+        vm.expectEmit(address(relic));
+        emit ShardBlacklistUpdated(RELIC_1_ID, shardIds[0], amounts[0]);
+        relic.unsetBlacklistedShards(bob, RELIC_1_ID, shardIds, amounts);
+
+        assertEq(relic.blacklistedAccounts(bob), true);
+        assertEq(relic.blacklistedShardsCount(shardIds[0]), 1);
+        assertEq(relic.blacklistedRelicShards(RELIC_1_ID, shardIds[0]), 1);
+
+        vm.expectEmit(address(relic));
+        emit ShardBlacklistUpdated(RELIC_1_ID, shardIds[0], amounts[0]);
+        relic.unsetBlacklistedShards(bob, RELIC_1_ID, shardIds, amounts);
+
+        assertEq(relic.blacklistedAccounts(bob), false);
+        assertEq(relic.blacklistedShardsCount(shardIds[0]), 0);
+        assertEq(relic.blacklistedRelicShards(RELIC_1_ID, shardIds[0]), 0);
+
+    }
+
+    function test_setRelicXP() public {
+        uint256[] memory relicIds = relic.relicsOfOwner(bob);
+        uint256 invalidRelicId = 999;
+        uint256 xp = 100;
+        vm.startPrank(executor);
+        vm.expectRevert(abi.encodeWithSelector(IRelic.InvalidRelic.selector, invalidRelicId));
+        relic.setRelicXP(invalidRelicId, xp);
+        uint256 relicId = relicIds[0];
+        IRelic.RelicInfoView memory relicInfoView = relic.getRelicInfo(relicId);
+        _setRarityThresholds();
+        assertEq(uint8(relicInfoView.rarity), uint8(IRelic.Rarity.Common));
+        
+        /// @dev commented out because expect emit oddly fails
+        // vm.expectEmit(address(relic));
+        // emit RelicXPSet(relicId, xp);
+        relic.setRelicXP(relicId, xp);
+
+        relicInfoView = relic.getRelicInfo(relicId);
+        assertEq(relicInfoView.xp, xp);
+        assertEq(uint8(relicInfoView.rarity), uint8(IRelic.Rarity.Legendary));
+    }
+}
+
+contract RelicViewTest is RelicSettersTest {
```

Comment:
Why is this inheriting RelicSettersTest? It only needs to inherit RelicTestBase right?

Impact is that there are duplicated test functions which are run

Reply:
You're right.